### PR TITLE
ability to create new exporter with options

### PR DESCRIPTION
**Pull request by Z10N0110**
the behavior of the exporter now controlled by options when new it, instead of only through the kingping flag

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -72,19 +66,48 @@ func (d *typedDesc) mustNewConstMetric(value float64, labels ...string) promethe
 	return prometheus.MustNewConstMetric(d.desc, d.valueType, value, labels...)
 }
 
-func NewExporter(address string, logger log.Logger) Exporter {
+type ExporterOption func(*Exporter)
```
Comment (SuperQ) [inline collector/collector.go:69]: I would prefer a config struct style of doing things. This is similar to how we're implementing this in other exporters.

```go
type ChronyCollectorConfig struct {
	CollectSource   bool
	CollectTracking bool
	...
}
```

Assemble the struct var in `main` and pass it to the `New` func.

**Review by SuperQ — CHANGES_REQUESTED** [review]

File: collector/collector.go (hunk 1 above)
Comment (Z10N0110) [reply collector/collector.go:69]: Ok, I will follow the style as the consul_exporter does.

**Review by SuperQ — APPROVED** [review]
Nice, thanks!

File: collector/collector.go [hunk 2]
```diff
@@ -72,16 +66,25 @@ func (d *typedDesc) mustNewConstMetric(value float64, labels ...string) promethe
 	return prometheus.MustNewConstMetric(d.desc, d.valueType, value, labels...)
 }
 
-func NewExporter(address string, logger log.Logger) Exporter {
+type ChronyCollectorConfig struct {
+	Address string
+	Timeout time.Duration
+
+	CollectSource      bool
```
Comment (SuperQ) [inline collector/collector.go:73]: Minor nit, naming mismatch.

```suggestion
	CollectSources     bool
```

**Comment by SuperQ** [comment]
I'm going to merge this and do a little var renaming before cutting a release.

**Pull request merged**